From 7a65fed3bc340195dc3dffd3df1f354df38dfaba Mon Sep 17 00:00:00 2001
From: QbProg <tholag@gmail.com>
Date: Sun, 24 Jul 2011 18:05:40 +0200
Subject: [PATCH 03/11] [warning-fix][missing-operators]

Added many placement delete operators, because the original classes didn't
have one (and the compiler complained)
---
 inc/BooleanOperations_AncestorsAndSuccessors.hxx   |    3 +++
 inc/BooleanOperations_ShapeAndInterferences.hxx    |    4 ++++
 inc/DBC_VArrayOfCharacter.hxx                      |    4 ++++
 inc/DBC_VArrayTNodeOfVArrayOfCharacter.hxx         |    4 ++++
 inc/DBC_VArrayTNodeOfVArrayOfExtCharacter.hxx      |    4 ++++
 inc/DBC_VArrayTNodeOfVArrayOfInteger.hxx           |    3 +++
 inc/DBC_VArrayTNodeOfVArrayOfReal.hxx              |    3 +++
 ...d_VArrayTNodeOfFieldOfHArray1OfBSplineCurve.hxx |    3 +++
 ...2d_VArrayTNodeOfFieldOfHArray1OfBezierCurve.hxx |    3 +++
 ...d_VArrayTNodeOfFieldOfHArray1OfBoundedCurve.hxx |    3 +++
 ...lPGeom2d_VArrayTNodeOfFieldOfHArray1OfCurve.hxx |    3 +++
 ...m_VArrayTNodeOfFieldOfHArray1OfBSplineCurve.hxx |    3 +++
 ...om_VArrayTNodeOfFieldOfHArray1OfBezierCurve.hxx |    3 +++
 ...m_VArrayTNodeOfFieldOfHArray1OfBoundedCurve.hxx |    3 +++
 ...VArrayTNodeOfFieldOfHArray1OfBoundedSurface.hxx |    3 +++
 ...ColPGeom_VArrayTNodeOfFieldOfHArray1OfCurve.hxx |    3 +++
 ...lPGeom_VArrayTNodeOfFieldOfHArray1OfSurface.hxx |    3 +++
 ...VArrayTNodeOfFieldOfHArray2OfBSplineSurface.hxx |    3 +++
 ..._VArrayTNodeOfFieldOfHArray2OfBezierSurface.hxx |    3 +++
 ...VArrayTNodeOfFieldOfHArray2OfBoundedSurface.hxx |    3 +++
 ...lPGeom_VArrayTNodeOfFieldOfHArray2OfSurface.hxx |    3 +++
 ...VArrayTNodeOfFieldOfHArray1OfExtendedString.hxx |    3 +++
 ...ColStd_VArrayTNodeOfFieldOfHArray1OfInteger.hxx |    3 +++
 ...Std_VArrayTNodeOfFieldOfHArray1OfPersistent.hxx |    3 +++
 inc/PColStd_VArrayTNodeOfFieldOfHArray1OfReal.hxx  |    3 +++
 ...ColStd_VArrayTNodeOfFieldOfHArray2OfInteger.hxx |    3 +++
 ...Std_VArrayTNodeOfFieldOfHArray2OfPersistent.hxx |    3 +++
 inc/PColStd_VArrayTNodeOfFieldOfHArray2OfReal.hxx  |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray1OfCirc2d.hxx |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray1OfDir.hxx    |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray1OfDir2d.hxx  |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray1OfLin2d.hxx  |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray1OfPnt.hxx    |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray1OfPnt2d.hxx  |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray1OfVec.hxx    |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray1OfVec2d.hxx  |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray1OfXY.hxx     |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray1OfXYZ.hxx    |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray2OfCirc2d.hxx |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray2OfDir.hxx    |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray2OfDir2d.hxx  |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray2OfLin2d.hxx  |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray2OfPnt.hxx    |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray2OfPnt2d.hxx  |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray2OfVec.hxx    |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray2OfVec2d.hxx  |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray2OfXY.hxx     |    3 +++
 inc/PColgp_VArrayTNodeOfFieldOfHArray2OfXYZ.hxx    |    3 +++
 inc/PDF_VArrayTNodeOfFieldOfHAttributeArray1.hxx   |    3 +++
 inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfByte.hxx |    3 +++
 ...rrayTNodeOfFieldOfHArray1OfHArray1OfInteger.hxx |    3 +++
 ..._VArrayTNodeOfFieldOfHArray1OfHArray1OfReal.hxx |    3 +++
 ...d_VArrayTNodeOfFieldOfHArray1OfHAsciiString.hxx |    3 +++
 ...ing_VArrayTNodeOfFieldOfHArray1OfNamedShape.hxx |    4 ++++
 ...PPoly_VArrayTNodeOfFieldOfHArray1OfTriangle.hxx |    3 +++
 ...hort_VArrayTNodeOfFieldOfHArray1OfShortReal.hxx |    3 +++
 ...hort_VArrayTNodeOfFieldOfHArray2OfShortReal.hxx |    3 +++
 ...PTopoDS_VArrayTNodeOfFieldOfHArray1OfHShape.hxx |    3 +++
 ...PTopoDS_VArrayTNodeOfFieldOfHArray1OfShape1.hxx |    3 +++
 inc/TopoDS_Iterator.hxx                            |    4 ++++
 inc/TopoDS_Shape.hxx                               |    3 +++
 61 files changed, 189 insertions(+), 0 deletions(-)

Index: oce/inc/BooleanOperations_AncestorsAndSuccessors.hxx
===================================================================
--- oce.orig/inc/BooleanOperations_AncestorsAndSuccessors.hxx
+++ oce/inc/BooleanOperations_AncestorsAndSuccessors.hxx
@@ -44,7 +44,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
   Standard_EXPORT   BooleanOperations_AncestorsAndSuccessors();
   //! allocates space and fills it with the data of AncSuccessors. <br>
Index: oce/inc/BooleanOperations_ShapeAndInterferences.hxx
===================================================================
--- oce.orig/inc/BooleanOperations_ShapeAndInterferences.hxx
+++ oce/inc/BooleanOperations_ShapeAndInterferences.hxx
@@ -58,6 +58,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
+
+  }
 
   
   Standard_EXPORT   BooleanOperations_ShapeAndInterferences();
Index: oce/inc/DBC_VArrayOfCharacter.hxx
===================================================================
--- oce.orig/inc/DBC_VArrayOfCharacter.hxx
+++ oce/inc/DBC_VArrayOfCharacter.hxx
@@ -55,6 +55,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
+
+  }
 
   
   Standard_EXPORT   DBC_VArrayOfCharacter();
Index: oce/inc/DBC_VArrayTNodeOfVArrayOfCharacter.hxx
===================================================================
--- oce.orig/inc/DBC_VArrayTNodeOfVArrayOfCharacter.hxx
+++ oce/inc/DBC_VArrayTNodeOfVArrayOfCharacter.hxx
@@ -42,6 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
+
+  }
 
   
       DBC_VArrayTNodeOfVArrayOfCharacter();
Index: oce/inc/DBC_VArrayTNodeOfVArrayOfExtCharacter.hxx
===================================================================
--- oce.orig/inc/DBC_VArrayTNodeOfVArrayOfExtCharacter.hxx
+++ oce/inc/DBC_VArrayTNodeOfVArrayOfExtCharacter.hxx
@@ -42,6 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
+
+  }
 
   
       DBC_VArrayTNodeOfVArrayOfExtCharacter();
Index: oce/inc/DBC_VArrayTNodeOfVArrayOfInteger.hxx
===================================================================
--- oce.orig/inc/DBC_VArrayTNodeOfVArrayOfInteger.hxx
+++ oce/inc/DBC_VArrayTNodeOfVArrayOfInteger.hxx
@@ -42,7 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       DBC_VArrayTNodeOfVArrayOfInteger();
   
Index: oce/inc/DBC_VArrayTNodeOfVArrayOfReal.hxx
===================================================================
--- oce.orig/inc/DBC_VArrayTNodeOfVArrayOfReal.hxx
+++ oce/inc/DBC_VArrayTNodeOfVArrayOfReal.hxx
@@ -42,7 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       DBC_VArrayTNodeOfVArrayOfReal();
   
Index: oce/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBSplineCurve.hxx
===================================================================
--- oce.orig/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBSplineCurve.hxx
+++ oce/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBSplineCurve.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBSplineCurve();
   
Index: oce/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBezierCurve.hxx
===================================================================
--- oce.orig/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBezierCurve.hxx
+++ oce/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBezierCurve.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBezierCurve();
   
Index: oce/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBoundedCurve.hxx
===================================================================
--- oce.orig/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBoundedCurve.hxx
+++ oce/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBoundedCurve.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfBoundedCurve();
   
Index: oce/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfCurve.hxx
===================================================================
--- oce.orig/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfCurve.hxx
+++ oce/inc/PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfCurve.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom2d_VArrayTNodeOfFieldOfHArray1OfCurve();
   
Index: oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBSplineCurve.hxx
===================================================================
--- oce.orig/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBSplineCurve.hxx
+++ oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBSplineCurve.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom_VArrayTNodeOfFieldOfHArray1OfBSplineCurve();
   
Index: oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBezierCurve.hxx
===================================================================
--- oce.orig/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBezierCurve.hxx
+++ oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBezierCurve.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom_VArrayTNodeOfFieldOfHArray1OfBezierCurve();
   
Index: oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBoundedCurve.hxx
===================================================================
--- oce.orig/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBoundedCurve.hxx
+++ oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBoundedCurve.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom_VArrayTNodeOfFieldOfHArray1OfBoundedCurve();
   
Index: oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBoundedSurface.hxx
===================================================================
--- oce.orig/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBoundedSurface.hxx
+++ oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfBoundedSurface.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom_VArrayTNodeOfFieldOfHArray1OfBoundedSurface();
   
Index: oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfCurve.hxx
===================================================================
--- oce.orig/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfCurve.hxx
+++ oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfCurve.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom_VArrayTNodeOfFieldOfHArray1OfCurve();
   
Index: oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfSurface.hxx
===================================================================
--- oce.orig/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfSurface.hxx
+++ oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray1OfSurface.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom_VArrayTNodeOfFieldOfHArray1OfSurface();
   
Index: oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfBSplineSurface.hxx
===================================================================
--- oce.orig/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfBSplineSurface.hxx
+++ oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfBSplineSurface.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom_VArrayTNodeOfFieldOfHArray2OfBSplineSurface();
   
Index: oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfBezierSurface.hxx
===================================================================
--- oce.orig/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfBezierSurface.hxx
+++ oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfBezierSurface.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom_VArrayTNodeOfFieldOfHArray2OfBezierSurface();
   
Index: oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfBoundedSurface.hxx
===================================================================
--- oce.orig/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfBoundedSurface.hxx
+++ oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfBoundedSurface.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom_VArrayTNodeOfFieldOfHArray2OfBoundedSurface();
   
Index: oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfSurface.hxx
===================================================================
--- oce.orig/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfSurface.hxx
+++ oce/inc/PColPGeom_VArrayTNodeOfFieldOfHArray2OfSurface.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColPGeom_VArrayTNodeOfFieldOfHArray2OfSurface();
   
Index: oce/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfExtendedString.hxx
===================================================================
--- oce.orig/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfExtendedString.hxx
+++ oce/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfExtendedString.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       PColStd_VArrayTNodeOfFieldOfHArray1OfExtendedString();
   
Index: oce/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfInteger.hxx
===================================================================
--- oce.orig/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfInteger.hxx
+++ oce/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfInteger.hxx
@@ -42,7 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       PColStd_VArrayTNodeOfFieldOfHArray1OfInteger();
   
Index: oce/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfPersistent.hxx
===================================================================
--- oce.orig/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfPersistent.hxx
+++ oce/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfPersistent.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       PColStd_VArrayTNodeOfFieldOfHArray1OfPersistent();
   
Index: oce/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfReal.hxx
===================================================================
--- oce.orig/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfReal.hxx
+++ oce/inc/PColStd_VArrayTNodeOfFieldOfHArray1OfReal.hxx
@@ -42,7 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       PColStd_VArrayTNodeOfFieldOfHArray1OfReal();
   
Index: oce/inc/PColStd_VArrayTNodeOfFieldOfHArray2OfInteger.hxx
===================================================================
--- oce.orig/inc/PColStd_VArrayTNodeOfFieldOfHArray2OfInteger.hxx
+++ oce/inc/PColStd_VArrayTNodeOfFieldOfHArray2OfInteger.hxx
@@ -42,7 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       PColStd_VArrayTNodeOfFieldOfHArray2OfInteger();
   
Index: oce/inc/PColStd_VArrayTNodeOfFieldOfHArray2OfPersistent.hxx
===================================================================
--- oce.orig/inc/PColStd_VArrayTNodeOfFieldOfHArray2OfPersistent.hxx
+++ oce/inc/PColStd_VArrayTNodeOfFieldOfHArray2OfPersistent.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       PColStd_VArrayTNodeOfFieldOfHArray2OfPersistent();
   
Index: oce/inc/PColStd_VArrayTNodeOfFieldOfHArray2OfReal.hxx
===================================================================
--- oce.orig/inc/PColStd_VArrayTNodeOfFieldOfHArray2OfReal.hxx
+++ oce/inc/PColStd_VArrayTNodeOfFieldOfHArray2OfReal.hxx
@@ -42,7 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       PColStd_VArrayTNodeOfFieldOfHArray2OfReal();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfCirc2d.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfCirc2d.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfCirc2d.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray1OfCirc2d();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfDir.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfDir.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfDir.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray1OfDir();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfDir2d.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfDir2d.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfDir2d.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray1OfDir2d();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfLin2d.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfLin2d.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfLin2d.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray1OfLin2d();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfPnt.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfPnt.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfPnt.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray1OfPnt();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfPnt2d.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfPnt2d.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfPnt2d.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray1OfPnt2d();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfVec.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfVec.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfVec.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray1OfVec();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfVec2d.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfVec2d.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfVec2d.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray1OfVec2d();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfXY.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfXY.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfXY.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray1OfXY();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfXYZ.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfXYZ.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray1OfXYZ.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray1OfXYZ();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfCirc2d.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfCirc2d.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfCirc2d.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray2OfCirc2d();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfDir.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfDir.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfDir.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray2OfDir();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfDir2d.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfDir2d.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfDir2d.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray2OfDir2d();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfLin2d.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfLin2d.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfLin2d.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray2OfLin2d();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfPnt.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfPnt.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfPnt.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray2OfPnt();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfPnt2d.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfPnt2d.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfPnt2d.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray2OfPnt2d();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfVec.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfVec.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfVec.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray2OfVec();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfVec2d.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfVec2d.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfVec2d.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray2OfVec2d();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfXY.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfXY.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfXY.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray2OfXY();
   
Index: oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfXYZ.hxx
===================================================================
--- oce.orig/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfXYZ.hxx
+++ oce/inc/PColgp_VArrayTNodeOfFieldOfHArray2OfXYZ.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PColgp_VArrayTNodeOfFieldOfHArray2OfXYZ();
   
Index: oce/inc/PDF_VArrayTNodeOfFieldOfHAttributeArray1.hxx
===================================================================
--- oce.orig/inc/PDF_VArrayTNodeOfFieldOfHAttributeArray1.hxx
+++ oce/inc/PDF_VArrayTNodeOfFieldOfHAttributeArray1.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PDF_VArrayTNodeOfFieldOfHAttributeArray1();
   
Index: oce/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfByte.hxx
===================================================================
--- oce.orig/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfByte.hxx
+++ oce/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfByte.hxx
@@ -42,7 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PDataStd_VArrayTNodeOfFieldOfHArray1OfByte();
   
Index: oce/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfHArray1OfInteger.hxx
===================================================================
--- oce.orig/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfHArray1OfInteger.hxx
+++ oce/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfHArray1OfInteger.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PDataStd_VArrayTNodeOfFieldOfHArray1OfHArray1OfInteger();
   
Index: oce/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfHArray1OfReal.hxx
===================================================================
--- oce.orig/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfHArray1OfReal.hxx
+++ oce/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfHArray1OfReal.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PDataStd_VArrayTNodeOfFieldOfHArray1OfHArray1OfReal();
   
Index: oce/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfHAsciiString.hxx
===================================================================
--- oce.orig/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfHAsciiString.hxx
+++ oce/inc/PDataStd_VArrayTNodeOfFieldOfHArray1OfHAsciiString.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PDataStd_VArrayTNodeOfFieldOfHArray1OfHAsciiString();
   
Index: oce/inc/PNaming_VArrayTNodeOfFieldOfHArray1OfNamedShape.hxx
===================================================================
--- oce.orig/inc/PNaming_VArrayTNodeOfFieldOfHArray1OfNamedShape.hxx
+++ oce/inc/PNaming_VArrayTNodeOfFieldOfHArray1OfNamedShape.hxx
@@ -43,6 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
+
+  }
 
   
       PNaming_VArrayTNodeOfFieldOfHArray1OfNamedShape();
Index: oce/inc/PPoly_VArrayTNodeOfFieldOfHArray1OfTriangle.hxx
===================================================================
--- oce.orig/inc/PPoly_VArrayTNodeOfFieldOfHArray1OfTriangle.hxx
+++ oce/inc/PPoly_VArrayTNodeOfFieldOfHArray1OfTriangle.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PPoly_VArrayTNodeOfFieldOfHArray1OfTriangle();
   
Index: oce/inc/PShort_VArrayTNodeOfFieldOfHArray1OfShortReal.hxx
===================================================================
--- oce.orig/inc/PShort_VArrayTNodeOfFieldOfHArray1OfShortReal.hxx
+++ oce/inc/PShort_VArrayTNodeOfFieldOfHArray1OfShortReal.hxx
@@ -42,7 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       PShort_VArrayTNodeOfFieldOfHArray1OfShortReal();
   
Index: oce/inc/PShort_VArrayTNodeOfFieldOfHArray2OfShortReal.hxx
===================================================================
--- oce.orig/inc/PShort_VArrayTNodeOfFieldOfHArray2OfShortReal.hxx
+++ oce/inc/PShort_VArrayTNodeOfFieldOfHArray2OfShortReal.hxx
@@ -42,7 +42,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   
       PShort_VArrayTNodeOfFieldOfHArray2OfShortReal();
   
Index: oce/inc/PTopoDS_VArrayTNodeOfFieldOfHArray1OfHShape.hxx
===================================================================
--- oce.orig/inc/PTopoDS_VArrayTNodeOfFieldOfHArray1OfHShape.hxx
+++ oce/inc/PTopoDS_VArrayTNodeOfFieldOfHArray1OfHShape.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PTopoDS_VArrayTNodeOfFieldOfHArray1OfHShape();
   
Index: oce/inc/PTopoDS_VArrayTNodeOfFieldOfHArray1OfShape1.hxx
===================================================================
--- oce.orig/inc/PTopoDS_VArrayTNodeOfFieldOfHArray1OfShape1.hxx
+++ oce/inc/PTopoDS_VArrayTNodeOfFieldOfHArray1OfShape1.hxx
@@ -43,7 +43,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void *)
+  {
 
+  }
   
       PTopoDS_VArrayTNodeOfFieldOfHArray1OfShape1();
   
Index: oce/inc/TopoDS_Iterator.hxx
===================================================================
--- oce.orig/inc/TopoDS_Iterator.hxx
+++ oce/inc/TopoDS_Iterator.hxx
@@ -53,6 +53,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void*)
+  {
+
+  }
 
   //! Creates an empty Iterator. <br>
       TopoDS_Iterator();
Index: oce/inc/TopoDS_Shape.hxx
===================================================================
--- oce.orig/inc/TopoDS_Shape.hxx
+++ oce/inc/TopoDS_Shape.hxx
@@ -63,7 +63,10 @@
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
 
+  }
   //! Creates a NULL Shape referring to nothing. <br>
       TopoDS_Shape();
   //! Returns true if this shape is null. In other words, it <br>
